embassy-mspm0/src/i2c.rs

```diff
@@ -0,0 +1,1156 @@
+#![macro_use]
+
+use core::future;
+use core::marker::PhantomData;
+use core::sync::atomic::{AtomicU32, Ordering};
+use core::task::Poll;
+
+use embassy_embedded_hal::SetConfig;
+use embassy_hal_internal::PeripheralType;
+use embassy_sync::waitqueue::AtomicWaker;
+use mspm0_metapac::i2c;
+
+use crate::gpio::{AnyPin, PfType, Pull, SealedPin};
+use crate::interrupt::typelevel::Binding;
+use crate::interrupt::{Interrupt, InterruptExt};
+use crate::mode::{Async, Blocking, Mode};
+use crate::pac::i2c::{vals, I2c as Regs};
+use crate::pac::{self};
+use crate::Peri;
+
+/// The clock source for the I2C.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum ClockSel {
+    /// Use the bus clock.
+    ///
+    /// By default the BusClk runs at 32 MHz.
+    BusClk,
```

Comment:
32MHz is probably correct on L130x? But G350x iirc has a PLL drive the bus clock?

I do want to support using the bus clock in the future. Although given the user could set the PLL in G350x to practically anything, maybe the BusClk variant would also pass the clock rate in Hertz? 

Before we do that we need to have a init do the proper clock configuration. For now we can  just support MfClk.

Reply:
Yep feel free to open an issue about clock config. I'll tag that issue when you do so.

Reply:
@i509VCB I changed the logic, clock source is defined by analyzing `bus_speed`  field of the config